finetuning/generalists/training/electron_microscopy/boundaries/obtain_boundaries_em_datasets.py

```diff
@@ -0,0 +1,147 @@
+    platy_cell_template = "membrane/train_data_membrane_%02i.n5"
+    platy_cell_label_key = "volumes/labels/segmentation/s1"
+
+    platy_cell_train_samples = [1, 2, 3, 4, 5, 6]
+    platy_cell_train_rois = compute_platy_rois(platy_root, platy_cell_train_samples, ignore_label=0,
+                                               file_template=platy_cell_template, label_key=platy_cell_label_key)
+    platy_cell_val_samples = [7, 8]
+    platy_cell_val_rois = compute_platy_rois(platy_root, platy_cell_val_samples, ignore_label=0,
+                                             file_template=platy_cell_template, label_key=platy_cell_label_key)
+
+    def cremi_dataset(rois):
+        return datasets.get_cremi_dataset(
+            path=os.path.join(input_path, "cremi"), patch_shape=patch_shape, label_transform=standard_label_trafo,
+            rois=rois, sampler=sampler, ndim=2, defect_augmentation_kwargs=None, download=True, raw_transform=identity
+        )
+
+    cremi_train_dataset = cremi_dataset(cremi_train_rois)
+    cremi_val_dataset = cremi_dataset(cremi_val_rois)
+
+    def platy_cell_dataset(rois, sample_ids):
+        return datasets.get_platynereis_cell_dataset(
+            path=platy_root, sample_ids=sample_ids, patch_shape=patch_shape, download=True, sampler=sampler,
+            ndim=2, rois=rois, raw_transform=ResizeRawTrafo(patch_shape[1:], do_rescaling=False),
+            label_transform=ResizeLabelTrafo(patch_shape[1:])
+        )
+
+    platy_cell_train_dataset = platy_cell_dataset(platy_cell_train_rois, platy_cell_train_samples)
+    platy_cell_val_dataset = platy_cell_dataset(platy_cell_val_rois, platy_cell_val_samples)
+
+    def axondeepseg_dataset(split):
+        return datasets.get_axondeepseg_dataset(
+            path=os.path.join(input_path, "axondeepseg"), name=["sem"], patch_shape=patch_shape[1:],
+            label_transform=axondeepseg_label_trafo, sampler=sampler, split=split,
+            raw_transform=identity, download=True, val_fraction=0.1
+        )
+
+    axondeepseg_train_dataset = axondeepseg_dataset("train")
+    axondeepseg_val_dataset = axondeepseg_dataset("val")
+
+    train_datasets = [cremi_train_dataset, platy_cell_train_dataset, axondeepseg_train_dataset]
+    val_datasets = [cremi_val_dataset, platy_cell_val_dataset, axondeepseg_val_dataset]
+
+    generalist_em_train_dataset = ConcatDataset(*train_datasets)
+    generalist_em_val_dataset = ConcatDataset(*val_datasets)
+
+    # creating weights for each concatenated dataset
+    train_weights = np.concatenate(
+        tuple([np.linspace(1, 3, len(_dataset)) for _dataset in train_datasets])
+    )
+    val_weights = np.concatenate(
+        tuple([np.linspace(1, 3, len(_dataset)) for _dataset in val_datasets])
+    )
+
+    train_sampler = WeightedRandomSampler(train_weights, len(generalist_em_train_dataset))
+    val_sampler = WeightedRandomSampler(val_weights, len(generalist_em_val_dataset))
```

Comment:
@constantinpape I scratched up an initlal setup for upsampling the AxonDeepSeg dataset. Works decent. (maybe you have another suggestion on this?)

Reply:
You could also just set `n_samples` to something higher in the axondeepseg dataset.
You can pass `n_samples` as kwarg to it, which will then be taken by the segmentation dataset:
https://github.com/constantinpape/torch-em/blob/main/torch_em/data/segmentation_dataset.py#L34

Reply:
It just repeats them.